lib/pagseguro/collection_base.rb

```diff
@@ -0,0 +1,41 @@
+module PagSeguro
+  class CollectionBase
+    extend CollectionManagers::MacroMethods
+    extend Forwardable
+    include Enumerable
+
+    def_delegators :@store, :size, :clear, :empty?, :any?, :each
+
+    def initialize
+      @store = []
+    end
+
+    def collection_manager
+      CollectionManagers::GenericManager
+    end
+
+    def collection_type
+      eval singularize(self.class.to_s)
```

Comment:
yes, but its returns me a String and I need a class...
I have tried with `Class.const_get` but not works in **ruby 1.9.3** and **ruby 2.0**.